src/EnvRegistry.ts

```diff
@@ -0,0 +1,49 @@
+import { from } from 'rxjs';
+import { switchMap } from 'rxjs/operators';
+import { messages } from '@capsulajs/capsulajs-configuration-service/lib/utils';
+import { ConfigurationService, ConfigurationServiceLocalStorage } from '@capsulajs/capsulajs-configuration-service';
+import {
+  EnvRegistry as EnvRegistryInterface,
+  EnvRegistryItem,
+  EnvironmentsRequest,
+  EnvironmentsResponse,
+  RegisterResponse
+} from './api/EnvRegistry';
+import { Env } from './api/Env';
+import { envKeyValidator, envValidator, validationMessages } from './utils';
+
+export default class EnvRegistry implements EnvRegistryInterface{
+  configurationService: ConfigurationService;
+  repositoryCreated: boolean;
+  repository: string;
+
+  constructor(private token: string) {
+    this.configurationService = new ConfigurationServiceLocalStorage(token); // add logic to choose specific provider
+    this.repositoryCreated = false;
+    this.repository = 'environmentRegistry';
+  }
+
+  private save(key: string, value: Env) {
```

Comment:
suggest to use object as an argument for consistency. You will be able to create an interface for it